### DISCUSSION TITLE
Adding a new unique key into a table with existing table

### DISCUSSION THREAD

**Discussion by jho14** [prisma/prisma#13284] [Q&A] [answered]
Hey Prisma community, 

I have an existing Prisma model with data being stored in Planetscale. I am attempting to add two new unique keys to the model, however Planetscale is throwing a fit (rightfully so.) Because those columns currently don't exist, each existing row will have to take on a default null which violates the @unique rule. I've come up with two workaround solutions below, but wanted to see if anyone had a better solution out there...this seems like it could be a common problem?

Original schema:
<img width="387" alt="Screen Shot 2022-05-10 at 6 30 16 PM" src="https://user-images.githubusercontent.com/60026007/167750775-03874e33-3d5b-4420-8b94-4e7c574675cb.png">

What we want:
<img width="411" alt="Screen Shot 2022-05-10 at 6 30 48 PM" src="https://user-images.githubusercontent.com/60026007/167750816-a426325a-2c5c-480d-8de2-83b15ea86176.png">

Workarounds:
1. Making two separate migrations.
    a) Default both url attributes to ("") and do not make them unique
    b) Write migrational code to update each row in the table with a unique string
    c) Once all rows have the unique key updated, change the schema so that the attributes are @unique
<img width="381" alt="Screen Shot 2022-05-10 at 6 38 32 PM" src="https://user-images.githubusercontent.com/60026007/167751536-6aedbcff-ce64-41d3-96d1-e1932a947d1b.png">

2. Create a new table
    a) Create a new table WaitlistURLs that only houses the urls and create a 1:1 relationship with WaitlistUser
<img width="833" alt="Screen Shot 2022-05-10 at 6 38 01 PM" src="https://user-images.githubusercontent.com/60026007/167751502-33bcf4d4-7bbf-464c-a255-a5fa2756e1d9.png">

Are there any better solutions out there that prisma offers? How have people solved this in plain MySQL?

Thanks a lot

**Comment by austincrim** [comment, marked as answer]
Hey @jho14 👋 ,

MySQL (which is what Planetscale uses, I believe) allows for multiple `NULL` values in a unique constraint. What errors are you getting from Planetscale?

**Comment by jho14** [comment]
This worked! Thanks @austincrim.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
